src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -235,18 +235,19 @@ public String getName() {
      * @return the version of the package
      * @since 1.0.0
      */
-    public String getVersion() {
+    public @Nullable String getVersion() {
         return version;
     }
 
     /**
      * Returns extra qualifying data for a package such as an OS, architecture, a distro, etc.
      * This method returns an UnmodifiableMap.
-     * @return qualifiers
+     *
+     * @return all the qualifiers, or an empty map if none are set
      * @since 1.0.0
      */
     public Map<String, String> getQualifiers() {
-        return (qualifiers != null) ? Collections.unmodifiableMap(qualifiers) : null;
+        return qualifiers != null ? Collections.unmodifiableMap(qualifiers) : Collections.emptyMap();
```

Comment:
Since one of the tests checks if all `PackageURL` getters return the same as the `PackageURLBuilder` getters, I introduced a behavioral change here too.